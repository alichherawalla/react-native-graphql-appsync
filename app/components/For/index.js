/**
 *
 * For
 *
 */

import React from 'react'
import Proptypes from 'prop-types'
import { StyleSheet, View } from 'react-native'

/**
 *
 */
export function For({
  of,
  ParentComponent = View,
  renderItem,
  flexDirection = 'column',
  marginLeft = 0
}) {
  const styles = StyleSheet.create({ parent: { flexDirection, marginLeft } })
  const list = () =>
    of.map((item, index) => ({ ...renderItem(item, index), key: index }))
  const children = () => (
    <ParentComponent testID="for" style={styles.parent}>
      {list()}
    </ParentComponent>
  )
  return (of || []).length ? children() : null
}

For.propTypes = {
  of: Proptypes.array,
  type: Proptypes.node,
  parent: Proptypes.object,
  iteratee: Proptypes.string,
  renderItem: Proptypes.func.isRequired
}
export default For
