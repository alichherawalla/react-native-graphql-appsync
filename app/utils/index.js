import { fonts } from 'app/themes'

export const getRandomKey = () =>
  Math.random()
    .toString(36)
    .substr(2, 3)
export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms))

export const rnFontSizeAdapter = type => ({
  fontSize: parseInt(
    fonts.size[type]()[0]
      .replace('font-size: ', '')
      .replace('px;', '')
      .trim(),
    10
  )
})

export const rnFontWeightAdapter = type => ({
  fontSize: parseInt(
    fonts.weights[type]()[0]
      .replace('font-weight: ', '')
      .trim(),
    10
  )
})

export const NEW_EMPLOYEE = {
  employee: {
    address: [{}],
    skills: [{}]
  }
}

export const SIGNUP_CONFIG = {
  header: 'Sign up to buildops',
  hideAllDefaults: true,
  defaultCountryCode: '1',
  signUpFields: [
    {
      label: 'User name',
      key: 'username',
      required: true,
      displayOrder: 1,
      type: 'string'
    },
    {
      label: 'Password',
      key: 'password',
      required: true,
      displayOrder: 2,
      type: 'password'
    },
    {
      label: 'Email',
      key: 'email',
      required: true,
      displayOrder: 4,
      type: 'string'
    }
  ]
}
