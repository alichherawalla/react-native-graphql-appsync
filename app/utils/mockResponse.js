export const employeeData = [
  {
    id: 1,
    firstname: 'John',
    lastname: 'Doe',
    address: [
      {
        line1: '3014 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3020 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 1, name: 'Technician' },
      { id: 2, name: 'Plumber' },
      { id: 3, name: 'Electrician' }
    ]
  },
  {
    id: 2,
    firstname: 'Jonathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [{ id: 6, name: 'Senior Electrician' }]
  },
  {
    id: 3,
    firstname: 'Steve',
    lastname: 'Doe',
    address: [
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' }
    ]
  },
  {
    id: 4,
    firstname: 'Nathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  },
  {
    id: 5,
    firstname: 'Albert',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  }
]
