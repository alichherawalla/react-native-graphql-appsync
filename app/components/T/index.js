/**
 *
 * T
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { injectIntl } from 'react-intl'
import { Text } from 'react-native-paper'
import { StyleSheet } from 'react-native'
import { rnFontSizeAdapter, rnFontWeightAdapter } from 'app/utils'

const T = ({ intl, id, values, text, size = 'small', weight = 'normal' }) => {
  const styles = StyleSheet.create({
    text: {
      ...rnFontSizeAdapter(size),
      ...rnFontWeightAdapter(weight)
    }
  })
  text = text || intl.formatMessage({ id }, { ...values })
  return (
    <Text testID="t" style={styles.text}>
      {text}
    </Text>
  )
}

T.propTypes = {
  id: PropTypes.string,
  intl: PropTypes.object,
  values: PropTypes.object,
  size: PropTypes.string,
  weight: PropTypes.string,
  text: PropTypes.string
}

export default compose(injectIntl)(T)
