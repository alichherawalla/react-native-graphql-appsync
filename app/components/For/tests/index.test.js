/**
 *
 * Tests for For
 *
 */

import React from 'react'
import { renderWithIntl } from 'app/utils/testUtils'
import T from 'app/components/T'
import For from '../index'

describe('<For />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(
      <For of={['test']} renderItem={() => <T text="s" />} />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 For component', () => {
    const { getAllByTestId } = renderWithIntl(
      <For of={['test']} renderItem={() => <T text="s" />} />
    )
    expect(getAllByTestId('for').length).toBe(1)
  })
})
