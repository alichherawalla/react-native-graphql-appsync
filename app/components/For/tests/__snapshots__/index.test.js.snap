// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<For /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "marginLeft": 0,
        }
      }
      testID="for"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Object {
              "fontSize": NaN,
            },
          ]
        }
        testID="t"
      >
        s
      </Text>
    </View>
  </View>
</View>
`;
